index.ts

```diff
@@ -351,28 +390,38 @@ export default class Analytics {
       retryOn: this.isErrorRetryable.bind(this),
     };
 
+    let error: Error | undefined = undefined;
     try {
       const response = await retryableFetch(`${this.host}`, req);
-      // handle success
-      if (response.ok) {
-        done();
-        return { data };
+      if (!response.ok) {
+        // handle 4xx 5xx errors
+        this.logger.error(
+          'request failed to send after 3 retries, dropping ' + itemsToFlush.length + ' events'
```

Comment:
what is the desired behavior when it comes to dropping events?

Reply:
afaik, you were for not dropping unless the call succeeded. Thanks for commenting here - I see the error old message is misleading and I should update it to match the code.